libs/logging/src/main/java/org/elasticsearch/logging/Logger.java

```diff
@@ -0,0 +1,718 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.logging;
+
+/**
+ * Main interface for logging. Most operations are done through this interface.
+ * This is interface heavily influenced by org.apache.logging.log4j2.Logger.
+ * The most notable difference is lack of methods with Marker, LogBuilder and
+ * the message supplier is {@code java.util.function.Supplier<String>}
+ */
+public interface Logger {
+    /**
+     * Logs a message String with the given level.
+     *
+     * @param level   the logging level
+     * @param message the message CharSequence to log.
+     */
+    void log(Level level, String message);
+
+    /**
+     * Logs a String message which is only to be constructed if the logging level is the specified level.
+     *
+     * @param level           the logging level
+     * @param messageSupplier A function, which when called, produces the desired log String message;
+     */
+    void log(Level level, java.util.function.Supplier<String> messageSupplier, Throwable throwable);
```

Comment:
Suggest, highlighting the LAZY aspect. e.g. 

"Logs a lazily supplied String message associated with a given throwable. If the logger is currently enabled for the specified log message level, then a message is logged that is the result produced by the given supplier function."

Note: missing at param for `throwable`